autopush_rs/src/util/ddb_helpers.rs

```diff
@@ -501,6 +566,241 @@ impl DynamoStorage {
         Box::new(response)
     }
 
+    fn drop_user(
+        ddb: Rc<Box<DynamoDb>>,
+        uaid: &Uuid,
+        router_table_name: &str,
+    ) -> MyFuture<DeleteItemOutput> {
+        let uaid = uaid.simple().to_string();
+        let table_name = router_table_name.to_string();
+        let response = retry_if(
+            move || {
+                ddb.delete_item(&DeleteItemInput {
+                    table_name: table_name.clone(),
+                    key: ddb_item! {
+                        uaid: s => uaid.clone()
+                    },
+                    ..Default::default()
+                })
```

Comment:
You can probably avoid some explicit captures outside this closure by creating the `DeleteItemInput` outside the closure and then moving it into the closure, although that's just a stylistic choice!